File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} candidates
     * @param {number} target
     * @return {number[][]}
     */
    combinationSum2(candidates, target) {
        const n = candidates.length
        candidates.sort((a, b) => a - b)

        const dfs = (acc, cur, sum, i) => {
            if (sum === target) return [...acc, cur]
            if (i >= n || sum > target) return acc

            const left = dfs(acc, [...cur, candidates[i]], sum + candidates[i], i + 1)
            let next_i = i + 1
            while (next_i < n && candidates[next_i - 1] === candidates[next_i]) next_i += 1
            const right = dfs(left, [...cur], sum, next_i)
            return right
        }

        const result = dfs([], [], 0, 0)
        return result
    }
}

File: Data Structures & Algorithms/non-overlapping-intervals/submission-0.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda v: v[1])

        cur = intervals[0][1]
        count = 1
        for interval in intervals[1:]:
            if interval[0] >= cur:
                cur = interval[1]
                count += 1

        return len(intervals) - count
